src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/changepolicy/ChangePolicyRequest.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.transport.action.changepolicy
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.model.ChangePolicy
+import org.elasticsearch.action.ActionRequest
+import org.elasticsearch.action.ActionRequestValidationException
+import org.elasticsearch.action.ValidateActions
+import org.elasticsearch.common.io.stream.StreamInput
+import org.elasticsearch.common.io.stream.StreamOutput
+import java.io.IOException
+
+class ChangePolicyRequest : ActionRequest {
+
+    val indices: List<String>
+    val changePolicy: ChangePolicy
+
+    constructor(
+        indices: List<String>,
+        changePolicy: ChangePolicy
+    ) : super() {
+        this.indices = indices
+        this.changePolicy = changePolicy
+    }
+
+    @Throws(IOException::class)
+    constructor(sin: StreamInput) : this(
+        indices = sin.readStringList(),
+        changePolicy = sin.readOptionalWriteable { ChangePolicy.fromStreamInput(it) } as ChangePolicy
```

Comment:
Is this optional? What does. request look like without a change policy?
Assuming it's not optional just change this to `changePolicy = ChangePolicy(sin)` now that you have a constructor.

Reply:
only have writeOptionalWriteable and readOptionalWriteable, so we need to use them,